app/workers/multiresimages_batch_worker.rb

```diff
@@ -9,46 +9,29 @@ class MultiresimagesBatchWorker
   include Sidekiq::Worker
 
   def perform(tiff_file)
-    # Regular expression swapping out file extension for .xml
-    xml = tiff_file.ext('xml')
-    raise 'XML file does not exist' unless File.exist?(xml)
+    raise 'XML file does not exist' unless File.exist?(tiff_file.ext('xml'))
 
-    doc = Nokogiri::XML(File.read( xml ))
-    accession_number = get_accession_number(doc)
+    nokogiri_doc = get_xml_doc(tiff_file)
+    accession_number = get_accession_number(nokogiri_doc)
 
-    raise "Invalid VRA" unless XSD.valid?(doc)
+    raise "Invalid VRA" unless XSD.valid?(nokogiri_doc)
     raise "No accession" if accession_number.empty?
     raise "Existing image found with this accession number: #{accession_number}" if Multiresimage.existing_image?(accession_number)
-    ready_xml = TransformXML.prepare_vra_xml(doc.to_xml)
+    ready_xml = TransformXML.prepare_vra_xml(nokogiri_doc.to_xml)
     pid = mint_pid("dil")
 
     begin
       logger.info("Batch worker starting - accession: #{accession_number}, pid: #{pid}")
-      m = nil
-      m = Multiresimage.new(pid: pid, vra_xml: ready_xml, from_menu: true)
-      m.save
-      # Copy tiff file to tmp directory
-      tmp_tiff_path = "tmp/#{m.tiff_img_name}"
-      tiff_derivative_path = m.tiff_derivative_path
-      FileUtils.cp(tiff_file, tmp_tiff_path)
-      m.create_datastreams_and_persist_image_files(tmp_tiff_path)
+      m = Multiresimage.create(pid: pid, vra_xml: ready_xml, from_menu: true)
+      m.create_datastreams_and_persist_image_files(tiff_file)
     rescue StandardError => e
-      unless m.nil? || m.destroyed?
-        m.delete
-      end
+      m.delete unless m.nil? || m.destroyed
+      File.unlink(m.tiff_derivative_path) if File.exist?(m.tiff_derivative_path)
```

Comment:
@csyversen I think this line will fail since m has just been deleted line above. Also is deleted in Multiresimage if there is a problem creating the datastreams.